pkg/kubelet/dockertools/manager.go

```diff
@@ -1261,33 +1271,46 @@ func (dm *DockerManager) KillPod(pod *api.Pod, runningPod kubecontainer.Pod) err
 				return
 			}
 
+			killContainerResult := &SyncResult{Action: KillContainer, Target: container.Name}
 			err := dm.KillContainerInPod(container.ID, containerSpec, pod, "Need to kill pod.")
 			if err != nil {
+				killContainerResult.Fail(kubecontainer.ErrKillContainer, err.Error())
 				glog.Errorf("Failed to delete container: %v; Skipping pod %q", err, runningPod.ID)
 				errs <- err
 			}
+			containerResults <- killContainerResult
 		}(container)
 	}
 	wg.Wait()
+	close(containerResults)
```

Comment:
I agree that `wg.Add(len(runningPod.Containers)` is better, but I didn't see any race in the current implementation. It seems that the link mainly said that we should not do `Wait` when the wg could be zero at first, or could be `Add` soon after becoming zero.